### PR TITLE
remove duplicate contexts - support array

### PR DESCRIPTION
**Pull request by gonenradai**
If top level object (Event, for example) has array of contexts and not only the single default context - these changes will remove any of the contexts that are already defined in parent from any child object.

based on discussion with @arwhyte - child objects are not expected to have array, thus the change leaves a child object context as-is and does not try to compare arrays.

### CODE REVIEW

**Comment by gonenradai** [comment]
Tested with the following cases:
* event has single context as string, all objects have the default context
* event has array of 2 contexts, all objects have default context
* event has array of 2 contexts, edApp (as an example) has array of 2 contexts
* event has array of 2 contexts, edApp (as an example) has array of 1 context 
* event has array of 2 contexts, edApp (as an example) has a string context which is not contained in event contexts

**Comment by lsloan** [comment]
@gonenradai, could you share some testing code?

File: src/util/ClassUtil.php [hunk 1]
```diff
@@ -52,7 +52,18 @@ protected function removeChildEntitySameContextsBase(array $serializationData, $
             if ($value instanceof \JsonSerializable) {
                 $value = $value->jsonSerialize();
                 if (is_array($value) && array_key_exists($contextProperty, $value)) {
-                    if ($value[$contextProperty] == $parent->getContext()) {
+                    if (is_array($value[$contextProperty]) && count($value[$contextProperty]) > 1) {
+                        // inner object has array of contexts - not trying to thin
+                        continue;
+                    } elseif (is_array($value[$contextProperty]) && count($value[$contextProperty]) === 1) {
```
Comment (lsloan) [inline src/util/ClassUtil.php]: Minor comment: `count()` always returns `int`, so the comparison operator doesn't need to be `===`.  It's fine to use `==`.

I'll give a more in-depth review and test the code soon.  If you can share some test cases, that would be helpful.

**Comment by gonenradai** [comment]
@lsloan 
as mentioned in the issue (#326 ) - in order to test there is a need to define a new class that extends Context and have its CONTEXT const be an array of few URLs.

The unit tests in this repo compare the output against fixed json files from the common-fixtures repository, so I can't modify a test which would actually pass a unit-test without also changing the corresponding fixed json file.

**Comment by lsloan** [comment]
@gonenradai 
I'll try to use the examples from the issue to see your change in action.  @arwhyte spoke with me about this PR and he seemed to think this change is ready to use.  So, I'm sorry if it seems like I am requesting unnecessary examples.  I want to make sure this change doesn't break existing code.

Although you're not able to change or add to the fixture files in their repo, you could email a file to me or attach it to a comment in this PR.

At the very least, does the code with your change pass its existing tests?

**Comment by gonenradai** [comment]
@lsloan 
here's a fixture I used
https://gist.github.com/gonenradai/ec1b6ec5cf57e74f97179546646361de
I duplicated the caliperEventMediaPausedVideo.json fixture and modified the top level context to be array of 2

here's the unit-test I created that with the modified ClassUtil should create a caliper event JSON that would match the fixture:
https://gist.github.com/gonenradai/2cc5a59910ae5f538e7ee86cf4df2328

running the tests:
* original, single context EventMediaPausedVideoTest.php - pass
* new, array of 2 contexts - EventMediaPausedVideoMultiContextTest.php - pass

let me know if there's anything else I can provide.

**Comment by lsloan** [comment]
I added the test case to this repo and the test fixture to IMSGlobal/caliper-common-fixtures. See PR: IMSGlobal/caliper-common-fixtures#214

**Pull request merged**